File: VB/03_Images/SetImageSize.vb
Imports Spire.Pdf
Imports Spire.Pdf.Graphics

Namespace SetImageSize
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Create a new PdfDocument object
            Dim doc As New PdfDocument()

            ' Add a new page to the document
            Dim page As PdfPageBase = doc.Pages.Add()

            ' Load an image from file
            Dim image As PdfImage = PdfImage.FromFile("..\..\..\..\..\..\Data\ChartImage.png")

            ' Calculate the width and height of the image with a scaling factor of 0.75
            Dim width As Single = image.Width * 0.75F
            Dim height As Single = image.Height * 0.75F

            ' Calculate the x and y coordinates to position the image at the center of the page
            Dim x As Single = (page.Canvas.ClientSize.Width - width) / 2
            Dim y As Single = 60.0F

            ' Draw the image on the page's canvas using the calculated positions and dimensions
            page.Canvas.DrawImage(image, x, y, width, height)

            ' Specify the name of the output file
            Dim result As String = "SetImageSize_out.pdf"

            ' Save the modified document to the specified file
            doc.SaveToFile(result)

            ' Close the document
            doc.Close()

            ' Launch the Pdf file
            PDFDocumentViewer(result)
        End Sub
        Private Sub PDFDocumentViewer(ByVal fileName As String)
            Try
                Process.Start(fileName)
            Catch
            End Try
        End Sub
    End Class
End Namespace

File: CS/03_Images/SetImageSize.cs
using Spire.Pdf;
using Spire.Pdf.Graphics;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace SetImageSize
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Create a pdf document.
            PdfDocument doc = new PdfDocument();

            //Create one page
            PdfPageBase page = doc.Pages.Add();

            //Load an image
            PdfImage image = PdfImage.FromFile(@"..\..\..\..\..\..\Data\ChartImage.png");

            //Set the width and height of image
            float width = image.Width * 0.75f;
            float height = image.Height * 0.75f;

            //Define a position to draw image
            float x = (page.Canvas.ClientSize.Width - width) / 2;
            float y = 60f;

            //Draw image on page canvas
            page.Canvas.DrawImage(image, x, y, width, height);

            String result = "SetImageSize_out.pdf";

            //Save the document
            doc.SaveToFile(result);

            //Launch the Pdf file
            PDFDocumentViewer(result);
        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}
